instrumentation/sidekiq/lib/opentelemetry/instrumentation/sidekiq/patches/poller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Sidekiq
+      module Patches
+        # The Poller module contains instrumentation for the enqueue and wait methods
+        module Poller
+          def enqueue
+            if config[:trace_poller_enqueue]
+              tracer.in_span('Sidekiq::Scheduled::Poller#enqueue') { super }
+            else
+              untraced { super }
+            end
+          end
+
+          private
+
+          def wait
+            if config[:trace_poller_wait]
+              tracer.in_span('Sidekiq::Scheduled::Poller#wait') { super }
+            else
+              untraced { super }
+            end
+          end
+
+          def tracer
+            Sidekiq::Instrumentation.instance.tracer
+          end
+
+          def config
+            Sidekiq::Instrumentation.instance.config
+          end
+
+          def untraced
+            OpenTelemetry::Trace.with_span(OpenTelemetry::Trace::Span.new) { yield }
+          end
```

Comment:
Is it worth extracting this to a common place?

Reply:
I think we could push this into the common util gem?

Reply:
https://github.com/open-telemetry/opentelemetry-ruby/pull/580